src/utils/database.ts

```diff
@@ -40,10 +49,66 @@ export async function dbFindOne<T>(
   filter: Filter<unknown>
 ): Promise<T | null> {
   try {
-    const result = await getCollection(collectionName).findOne(filter);
-    return result as T | null;
+    const result = await getCollection<T>(collectionName).findOne(filter);
+    return result;
   } catch (error) {
     console.error(`DB Error in dbFindOne: ${error} --- Filter: ${filter}`);
     return null;
   }
 }
+
+export async function dbFind<T>(
+  collectionName: string,
+  filter: Filter<unknown>,
+  options: { sortKey?: string; sortDirection?: SortDirection; limit?: number } = {}
+): Promise<T[] | null> {
+  try {
+    const findResult = getCollection<T>(collectionName).find(filter);
+    const sortedResult = options.sortKey
+      ? findResult.sort(options.sortKey, options.sortDirection)
+      : findResult;
+    const limitedResult = options.limit ? sortedResult.limit(options.limit) : sortedResult;
+    const result = await limitedResult.toArray();
+    return result;
+  } catch (error) {
+    console.error(`DB Error in dbFind: ${error} --- Filter: ${filter}`);
+    return null;
```

Comment:
I don't think catching an exception and then only logging it is the best way, since it isn't really handling the error. Wouldn't it be possible to return the error?

Reply:
As explained, the null return is handled in the calling function where a 400/500 response is created.
I don't like to pass the error message to the frontend due to security reasons.